pkg/project/auth/cache.go

```diff
@@ -301,21 +246,27 @@ func purgeDeletedNamespaces(namespaceSet *util.StringSet, userSubjectRecordStore
 func (ac *AuthorizationCache) invalidateCache() bool {
 	invalidateCache := false
 
-	clusterPolicies := ac.clusterPolicyStore.List()
-	for i := range clusterPolicies {
-		policy := clusterPolicies[i].(*authorizationapi.ClusterPolicy)
-		if policy.ResourceVersion != ac.clusterPolicyResourceVersion {
+	clusterPolicyList, err := ac.policyClient.ReadOnlyClusterPolicies().List(labels.Everything(), fields.Everything())
+	if err != nil {
+		util.HandleError(err)
+		return invalidateCache
+	}
+	for _, clusterPolicy := range clusterPolicyList.Items {
```

Comment:
so add each resourceVersion of a clusterPolicy into a local set, and then compare the local set with the old set, if they are not the same, we invalidate the cache.